GeoLib/Grid.h

```diff
@@ -201,6 +201,10 @@ Grid<POINT>::Grid(InputIterator first, InputIterator last,
                                     _max_pnt[1] - _min_pnt[1],
                                     _max_pnt[2] - _min_pnt[2]}};
 
+    // enlarge delta
+    for (auto & d : delta)
+        d = std::nextafter(d, std::numeric_limits<double>::max());
```

Comment:
> I think the result of std::nextafter can be cancelled out by very many arithmetic operations (e.g. adding or subtracting a large number). 

In general you re right. The bounding box is an half open interval. As you pointed out already out the arithmetic operations computing `delta` cancel out the results of `std::nextafter` in the bounding box. For this reason I reenlarge `delta`.

 > I think multiplying with 1+eps would be more robust ...

Sometimes all points have the same z coordinate, i.e., the `delta[2] == 0`. So the half open interval has to be enlarged. In this case the multiplication with `1+eps` does not lead to the desired solution.